src/Pim/Bundle/CatalogBundle/Resources/config/validation/product.yml

```diff
@@ -17,11 +17,12 @@ Pim\Component\Catalog\Model\Product:
                 - pim_catalog_variant_group
     properties:
         identifier:
-            # TODO: TIP-722 - Add not Blank and Length validation here once the product value
-            # identifier is removed.
+            - NotBlank: ~
```

Comment:
unfortunately this may break the API Violations endpoint (see ViolationNormalizer). :/